src/interfaces/records/handlers/records-write.ts

```diff
@@ -77,21 +79,26 @@ export class RecordsWriteHandler implements MethodHandler {
     const indexes = await constructRecordsWriteIndexes(recordsWrite, isLatestBaseState);
 
     try {
-      this.validateUndefinedDataStream(dataStream, newestExistingMessage, message);
-
-      await this.storeMessage(this.messageStore, this.dataStore, this.eventLog, tenant, message, indexes, dataStream);
+      // try to store data, unless options explicitly say to skip storage
+      if (!options?.skipDataStorage) {
```

Comment:
Optional nit: I find it a bit confusing with the double-negative...

```suggestion
      if (options === undefined || !options.skipDataStorage) {
```